content/docs/dvclive/monitor_system.md

```diff
@@ -0,0 +1,166 @@
+# Monitoring System Metrics
+
+DVCLive can track and log your system metrics so you can see your hardware
+performance and limitations during your experiments. The metrics concerns your
+GPUs usage, if you have some, and your CPU and RAM memory usage. It can also
+monitor the disk usage.
+
+## Usage
+
+To monitor your system metrics you need to install the optional dependencies
+first:
+
+```cli
+$ pip install "dvclive[system]"
+```
+
+Then you can set the `monitor_system` parameter to `True` in `Live`:
+
+```py
+from dvclive import Live
+
+live = Live(monitor_system=True)
+```
+
+or you can specify additional parameters to configure the logger behavior:
+
+```py
+from dvclive import Live
+from dvclive.monitor_system import SystemMonitor
+
+live = Live()
+live.monitor_system = SystemMonitor(
+    interval = 0.01
+    num_samples = 20,
+    directories_to_monitor = {"data": "/data", "users": "/home"},
+    plot = True,
+)
+```
+
+<toggle>
+<tab title="DVC Studio">
+
+![DVCLive Studio System Metrics Plots](/img/dvclive-studio-system-metrics-plots.png)
+![DVCLive Studio System Metrics](/img/dvclive-studio-system-metrics.png)
+
+</tab>
+<tab title="VSCode Extension">
+
+![DVCLive VSCode System Metrics Plots](/img/dvclive-vscode-system-metrics-plots.png)
+![DVCLive VSCode System Metrics](/img/dvclive-vscode-system-metrics.png)
+
+</tab>
+
+</toggle>
+
+## Description
+
+The `SystemMonitor` class is used to configure the system logger. If you call
+`Live` with `monitor_system=True` it will create a `SystemMonitor` with default
+parameters. It works by sampling your system metrics at regular intervals and
+average them over a number of samples. The averaged data is then
+[logged as a metric](https://dvc.org/doc/dvclive/live/log_metric).
+
+The `SystemMonitor` has the following parameters:
+
+```py
+class SystemMonitor:
+
+    def __init__(
+        self,
+        interval: float = 0.05,
+        num_samples: int = 20,
+        directories_to_monitor: Optional[Dict[str, str]] = None,
+        plot: bool = True,
+        ):
+```
+
+- `interval` - the time interval between samples in seconds. To keep the
+  sampling interval small, the maximum value allowed is 0.1 seconds. Default to
+  0.05.
+- `num_samples` - the number of samples to collect before the aggregation. The
+  value should be between 1 and 30 samples. Default to 20.
+- `directories_to_monitor` - a dictionary with the information about which
+  directories to monitor. The `key` would be the name of the metric and the
+  `value` is the path to the directory. The metric tracked concerns the
+  partition that contains the directory. Default to `{"main": "/"}`.
```

Comment:
Why require a partition/dir name instead of using list like `"/"` and using `disk_path.strip("/")` as the name? Just curious if we can simplify this argument.

Reply:
I know some use cases where your partition is mounted on the long path (also, it is easy to reach long paths if you're using docker volumes), and I didn't want the metric name to be 20 characters long.
Also, with the metrics name as a `key`, users can change the path of their data (during code refacto, for instance) without changing the name of the metrics. With this, it will be easy to compare new experiments with old ones, even if the path to your data changes. Let's say I dockerize my code to run experiments; the path to my images will change from `/home/alex/project/data/` to `/usr/images`, for instance. If you specify the name you want for the metrics, your dashboard will look the same before and after.

But I agree, this `directories_to_monitor` looks difficult to grasp. 

One solution is to accept lists and dictionaries. But I fear that it will further complicate the argument. What do you think?

Reply:
Okay, not a blocker. It's fine to keep it as is. Thanks for the explanation!